### PR TITLE
Add bounds check to generic number parsing

### PR DESCRIPTION
**Pull request by lask**

### CODE REVIEW

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/core/numbers.toit [hunk 1]
```diff
@@ -618,19 +620,38 @@ abstract class int extends num:
     else:
       return parse_generic_radix_ radix data from to --on_error=on_error
 
+  static char_to_int_ c/int -> int:
```
Comment (erikcorry) [inline lib/core/numbers.toit:623]: 7 lines up the condition in parse_ should probably be
if not 0 <= from <= to <= data.size

File: lib/core/numbers.toit [hunk 2]
```diff
@@ -618,19 +620,38 @@ abstract class int extends num:
...
     if not 2 <= radix <= 36: throw "INVALID_RADIX"
 
-    max_num := (min radix 9) + '0' - 1
-    max_char := radix + 'a' - 1
-    max_char_C := radix + 'A' - 1
-    return generic_parser_ data from to --on_error=on_error: | char result _ _ |
+    max_num := (min radix 10) + '0' - 1
+    max_char := radix - 10 + 'a' - 1
+    max_char_C := radix - 10 + 'A' - 1
+    max_int64_div_radix := (data.size > 12) ? MAX / radix : MAX
```
Comment (erikcorry) [inline lib/core/numbers.toit:635]: data.size should be from - to

**Review by erikcorry — APPROVED** [review]

File: lib/core/numbers.toit (hunk 1 above)
Comment (erikcorry) [reply lib/core/numbers.toit:623]: or
`if not 0 <= from < to <= data.size`

**Pull request merged**